### PR TITLE
Initial Code organization and dockerfiles for STAR and CellRanger 

### PR DESCRIPTION
**Pull request by wpoehlm**

### CODE REVIEW

**Comment by tthyer** [comment]
The organization looks good. I have two suggestions.

First, I would rename "Unit_Test" to just "test" -- that's what is conventional and expected. I still think having "functional_test" in the tool subfolders is fine / makes the distinction clear. 

Second, try to pick one naming syntax and stick with that. It is most common to see lower case names, again conventional and expected. If you prefer to go with upper case ones, be consistent, i.e. don't have "Unit_Test" and "functional_test", either lowercase all or titlecase all. Also, There is a mix of camelcase (CellRanger) and title+snakecase (Unit_Test).

**Comment by wpoehlm** [comment]
> 
> 
> The organization looks good. I have two suggestions.
> 
> First, I would rename "Unit_Test" to just "test" -- that's what is conventional and expected. I still think having "functional_test" in the tool subfolders is fine / makes the distinction clear.
> 
> Second, try to pick one naming syntax and stick with that. It is most common to see lower case names, again conventional and expected. If you prefer to go with upper case ones, be consistent, i.e. don't have "Unit_Test" and "functional_test", either lowercase all or titlecase all. Also, There is a mix of camelcase (CellRanger) and title+snakecase (Unit_Test).

Good suggestions!  I cleaned up the folder names for consistency

File: tools/cell_ranger/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+# export cargo path    
+ENV PATH="/root/.cargo/bin:$PATH"
+
+# install correct version of rustup
+RUN rustup install 1.28.0 && rustup default 1.28.0
+
+# clone cellranger sourcecode repo, cd into this dir, and install cellranger
+RUN git clone https://github.com/10XGenomics/cellranger \
+  && cd cellranger && make
```
Comment (tthyer) [inline tools/cell_ranger/Dockerfile]: How did building cellranger go? If ok, why is the gzipped cellranger in .gitignore?

**Review by tthyer — APPROVED** [review]

File: tools/cell_ranger/Dockerfile (hunk 1 above)
Comment (wpoehlm) [reply tools/cell_ranger/Dockerfile]: It seemed to build fine, however missing dependencies are coming up when actually running it with real data.  Still troubleshooting but I guess we have the tarball as a backup option

**Pull request merged**